helpers/sql-obfuscation/lib/opentelemetry/helpers/sql_obfuscation.rb

```diff
@@ -127,6 +129,8 @@ def obfuscate_sql(sql, obfuscation_limit: 2000, adapter: :default)
 
       # @api private
       def truncate_statement(sql, regex, limit)
+        sql = sql.gsub(PREPENDED_COMMENT_REGEX, PLACEHOLDER) if sql.match?(PREPENDED_COMMENT_REGEX)
```

Comment:
@kaylareopelle I think what we need here is to obfuscate the substring and amend the truncation message. 

Something like 

```ruby

if sql.length >= limit
  sql = sql[0..limit-1] + "SQL truncated (> #{limit} characters)"
end

sql.gsub(regex)
```

Reply:
I'll try it out!

Reply:
@arielvalentin - With this approach, if the limit is hit in the middle of a value that needs to be obfuscated, the obfuscation will fail for that value. 

The string will miss the closing part of the capture group, so the value won't be replaced with the placeholder.

This can be prevented by running the query through the CLEANUP_REGEX before sending it out, but every time the query was truncated within a sensitive value, `sql.statement` would equal `'Failed to obfuscate SQL query - quote characters remained after obfuscation'`.

I'll keep working on this tomorrow to see if there's another way around this problem, let me know if you have other ideas!

Just to clarify what we're working toward, it sounds like the ideal situation would be
* truncate the query to the limit
* obfuscate everything within that truncated string

Does that sound right?

Reply:
I suppose. 

I think if this gets too complicated, I think we should highly consider not processing the SQL at all and return the placeholder value.

I think we may also want to consider increasing the size limit. 

I don't know why we landed in 2K characters but it seems arbitrary. 

Maybe we publish benchmarks or something to give folks an idea of what to expect? 

IDK.

Reply:
It looks like the 2000 limit [has been around since the beginning.](https://github.com/open-telemetry/opentelemetry-ruby-contrib/commit/ccfd3d19194bb8b55e93e96adfdd600e28b30637)

Here's the PR that introduced the current truncation style: https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/224 
[NR's limit is 16384](https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent/database.rb#L12). So we could consider changing the limit.

I like the idea of publishing benchmarks. When we decide what we want to do with the truncated queries, I'll add one that toggles the various configs.

> I think if this gets too complicated, I think we should highly consider not processing the SQL at all and return the placeholder value.

I agree. I'm still experimenting with different strategies, but have this timeboxed.

Reply:
I love how when we look at old PRs, I was the approver 🤦🏼 

I only have myself to blame.

Reply:
The original, original, original one 

https://github.com/aislinnnnn/opentelemetry-ruby/commit/1bd5e24fd4b7346b281dbb72d9b34dbab360f361

There isn't much context on the size choice. 

> [NR's limit is 16384](https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent/database.rb#L12). So we could consider changing the limit.

Agreed.

Reply:
It looks like for OTel, [NR limits attribute values to 4095](https://github.com/newrelic/docs-website/blob/develop/src/content/docs/opentelemetry/best-practices/opentelemetry-otlp.mdx?plain=1#L209). So 4000 might be a better number here? 🤷

I can check in with other SIG members next week to see if they have suggestions on a reasonable limit.